examples/ConstraintIB/eel2d/example.cpp

```diff
@@ -261,26 +308,101 @@ run_example(int argc, char* argv[])
         // Main time step loop.
         double loop_time_end = time_integrator->getEndTime();
         double dt = 0.0;
+        double current_time, new_time;
+        double box_disp = 0.0;
         while (!MathUtilities<double>::equalEps(loop_time, loop_time_end) && time_integrator->stepsRemaining())
         {
             iteration_num = time_integrator->getIntegratorStep();
             loop_time = time_integrator->getIntegratorTime();
+            current_time = loop_time;
 
             pout << "\n";
             pout << "+++++++++++++++++++++++++++++++++++++++++++++++++++\n";
             pout << "At beginning of timestep # " << iteration_num << "\n";
             pout << "Simulation time is " << loop_time << "\n";
 
             dt = time_integrator->getMaximumTimeStepSize();
-            time_integrator->advanceHierarchy(dt);
             loop_time += dt;
+            new_time = loop_time;
+
+            // Regrid the hierarchy if necessary
```

Comment:
Yes you are certainly going to want to compute things on the same hierarchy. Could you "inject" the code into the time step loop as a postprocessing routine registered with the hierarchy integrator, so that you have access to u(n) and u(n+1) on the same hierarchy that they were computed on?